src/Microsoft.AspNet.OData/Adapters/WebApiContext.cs

```diff
@@ -0,0 +1,119 @@
+﻿// Copyright (c) Microsoft Corporation.  All rights reserved.
+// Licensed under the MIT License.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.AspNet.OData.Common;
+using Microsoft.AspNet.OData.Extensions;
+using Microsoft.AspNet.OData.Interfaces;
+using Microsoft.AspNet.OData.Routing.Conventions;
+using Microsoft.OData.UriParser;
+using Microsoft.OData.UriParser.Aggregation;
+using ODataPath = Microsoft.AspNet.OData.Routing.ODataPath;
+
+namespace Microsoft.AspNet.OData.Adapters
+{
+    /// <summary>
+    /// Adapter class to convert Asp.Net WebApi OData properties to OData WebApi.
+    /// </summary>
+    internal class WebApiContext : IWebApiContext
+    {
+        /// <summary>
+        /// The inner context wrapped by this instance.
+        /// </summary>
+        private HttpRequestMessageProperties innerContext;
+
+        /// <summary>
+        /// Initializes a new instance of the WebApiContext class.
+        /// </summary>
+        /// <param name="context">The inner context.</param>
+        public WebApiContext(HttpRequestMessageProperties context)
+        {
+            if (context == null)
+            {
+                throw Error.ArgumentNull("context");
+            }
+
+            this.innerContext = context;
+        }
+
+        /// <summary>
+        /// Gets or sets the parsed OData <see cref="ApplyClause"/> of the request.
+        /// </summary>
+        public ApplyClause ApplyClause
+        {
+            get { return this.innerContext.ApplyClause; }
+            set { this.innerContext.ApplyClause = value; }
+        }
+
+        /// <summary>
+        /// Gets or sets the next link for the OData response.
+        /// </summary>
+        public Uri NextLink
+        {
+            get { return this.innerContext.NextLink; }
+            set { this.innerContext.NextLink = value; }
+        }
+
+        /// <summary>
+        /// Gets or sets the delta link for the OData response.
+        /// </summary>
+        public Uri DeltaLink
+        {
+            get { return this.innerContext.DeltaLink; }
+            set { this.innerContext.DeltaLink = value; }
+        }
+
+        /// <summary>
+        /// Gets or sets the OData path.
+        /// </summary>
+        public ODataPath Path
+        {
+            get { return this.innerContext.Path; }
+        }
+
+        /// <summary>
+        /// Gets the route name for generating OData links.
+        /// </summary>
+        public string RouteName
+        {
+            get { return this.innerContext.RouteName; }
+        }
+
+        /// <summary>
+        /// Gets the data store used by <see cref="IODataRoutingConvention"/>s to store any custom route data.
+        /// </summary>
+        /// <value>Initially an empty <c>IDictionary&lt;string, object&gt;</c>.</value>
```

Comment:
Text encoding/decoding hiccup?

Reply:
Note, that is intended to produce IDictionary<string, object> but you have to escape the <>'s in comments.